compiler-core/src/language_server/engine.rs

```diff
@@ -80,6 +84,11 @@ where
         })
     }
 
+    /// Compiles the current project, returning a `Response` with any warnings and the
+    /// paths of any modules that were compiled.
+    ///
+    /// If there is no project, or the project is not a Gleam project, then
+    /// nothing is compiled.
```

Comment:
```suggestion
```

Not quite! Each engine always has a project. If the request is for a file outside a project then there is no engine.